doc/performance_blog_post/performance_blog_post.md

```diff
@@ -0,0 +1,109 @@
+
+## Writing Performance Tests
+
+So my first step was to write automated performance tests for the parquet loading. Don't get this wrong. It's not that we did not have any tests before. Of course, we had unit and integration tests for the Virtual Schema that make sure that the Virtual Schema works as expected. But since this is a rather new product we did not have automated performance tests yet.
+
+Until now, we checked the performance of this Virtual Schema manually by starting a cluster and querying a parquet dataset. However, for these performance improvements we needed more test cases and also more reliability.
+
+While for our integration tests we use the docker version of Exasol (which is absolutely sufficient for functional testing), for the performance test we want to test with a real-live setup. So we decided to test with a 4 node cluster (with c5.4xlarge instances) on AWS. There we load the files from S3 using the [s3-document-files-virtual-schema](https://github.com/exasol/s3-document-files-virtual-schema).
+
+The virtual Schemas are built out of multiple layers:
+
+![Stack of virtual schema layers](vsDocStack.png)
+
+The parquet loading implementation is located in the Document Files Virtual Schema. However, we can't test this directly. Instead, we decided to test the S3 implementation, since this is the most popular dialect.
+
+### Test Quality
+
+We measure the query performance by measuring the execution time of a query on a Virtual Schema table. By that we measure the full execution time including network transmission, S3 loading and so on. This of course adds some noise to our measurement. However, we decided for this strategy since it measures the time that our customers can expect.
+
+In order to make the tests more reliable we run each test 5 times. Then we drop the first 2 results, since they are usually slower, since S3 will keep the files in cache for the following runs. Then we build the average of the remaining 3 runs. We decided for the average since it will again result in a value that customers can expect.
+
+### Test Fixtures
+
+For our performance test we need some test fixtures. For example one big parquet file on S3 that we load with a Virtual Schema and measure the performance.
+
+The simplest option for this is to create this file with some tool locally and store it forever on S3. This approach has the advantage, that its easy to create. On the other hand, it's hard to change, and it's also not apparent from the source code how the test data is created. Since out projects are open source, its also important that all (non Exasol-) contributors can run the tests. So we would need to make our S3 bucket publicly available which could cause costs.
```

Comment:
```suggestion
The simplest option for this is to create this file with local tool and store it forever on S3. This approach has the advantage, that it's easy to create. On the other hand, it's hard to change, and it's also not apparent from the source code how the test data is created. Since out projects are open source, its also important that all external contributors can run the tests. This however would require us to make our S3 bucket publicly available which could cause additional costs.
```